src/main/scala/org/apache/sysml/compiler/lang/source/DML.scala

```diff
@@ -0,0 +1,473 @@
+
+            (target, argss) match {
+
+              /* matches tuples */
+              case (Some(tgt), _) if isApply(method) && api.Sym.tuples.contains(method.owner.companion) =>
+                ""
+
+              case (Some(tgt), _) if method == api.Sym.foreach || method.overrides.contains(api.Sym.foreach) =>
+                forLoop(tgt, targs, argss.flatten, env)
+
+              /* matches unary methods without arguments, e.g. A.t */
+              case (Some(tgt), Nil) if isUnary(method) =>
+                s"${printSym(method)}${tgt(env)}"
+
+              /* matches apply methods with one argument */
+              case (Some(tgt), ((arg :: Nil) :: Nil)) if isApply(method) => {
+                val module = tgt(env)
+
+                if (module == "Vector") { // Vector.apply(Array(...))
+                  printConstructor(method, argss, env, true)
+                }
+
+                else {
+                  s"${tgt(env)}${printMethod(" ", method, " ")}${arg(env)}"
+                }
+              }
+
+              /* matches methods with one argument (%*%, read) */
+              case (Some(tgt), (arg :: Nil) :: Nil) => {
+
+                val module = tgt(env)
+
+                // methods in the package object (builtins with one argument (read)
+                if (module == "package") {
+                  printBuiltin(tgt, method, argss, env)
+                }
+
+                // matrix constructors with one argument (fromDataFrame)
+                else if (module == "Matrix") {
+                  printConstructor(method, argss, env, false)
+                }
+
+                else if (module == "Vector") {
+                  printConstructor(method, argss, env, true)
+                }
+
+                // methods from scala.predef with one argument (println(...) etc.)
+                else if (module == "Predef") {
+                  val name: u.TermName = method.name.decodedName.toTermName
+
+                  name match {
+                    case u.TermName(tn) if tn == "println" || tn == "print" => s"print(${arg(env)})"
+                    case u.TermName("intWrapper") => s"${arg(env)}"
+                    case _ => abort(s"scala.predef.$name is not supported in DML", method.pos)
+                  }
+                }
+
+                // binary operators
+                else {
+                  method.name.decodedName.toTermName match {
+                    case u.TermName("to")    => s"${tgt(env)} + 1:${arg(env)} + 1"
+                    case u.TermName("until") => s"${tgt(env)} + 1:${arg(env)}"
+                    case u.TermName("%") => s"($module %% ${args(0)})" // modulo in dml is %%
+                    case u.TermName("&&") => s"($module & ${args(0)})" // && in dml is &
+                    case u.TermName("||") => s"($module | ${args(0)})" // || in dml is |
+                    case _ => s"($module ${method.name.decodedName} ${args(0)})"
```

Comment:
Is this default cause guaranteed not to generate nonsensical DML, or should there be some additional error-checking code here?